### PR TITLE
Addition of external_app

### PR DESCRIPTION
**Pull request by frkv**

### CODE REVIEW

File: nordicsemi/dfu/package.py [hunk 1]
```diff
@@ -371,7 +382,7 @@ def generate_package(self, filename, preserve_work_dir=False):
             sd_size = 0
             bl_size = 0
             app_size = 0
-            if key == HexType.APPLICATION:
+            if key == HexType.APPLICATION or key == HexType.EXTERNAL_APPLICATION:
```
Comment (oyvindronningstad) [inline nordicsemi/dfu/package.py]: Could be:

    if key in [HexType.APPLICATION, HexType.EXTERNAL_APPLICATION]

File: nordicsemi/dfu/package.py [hunk 2]
```diff
@@ -156,10 +159,17 @@ def __init__(self,
         self.firmwares_data = {}
 
         if app_fw:
-            self.__add_firmware_info(firmware_type=HexType.APPLICATION,
-                                     firmware_version=app_version,
-                                     filename=app_fw,
-                                     init_packet_data=init_packet_vars)
+            if is_external:
```
Comment (oyvindronningstad) [inline nordicsemi/dfu/package.py]: Could be:

            firmware_type = HexType.EXTERNAL_APPLICATION if is_external else HexType.APPLICATION
            self.__add_firmware_info(firmware_type=firmware_type,
                                     firmware_version=app_version,
                                     filename=app_fw,
                                     init_packet_data=init_packet_vars)

File: nordicsemi/dfu/package.py (hunk 1 above)
Comment (frkv) [reply nordicsemi/dfu/package.py]: Changing it

File: nordicsemi/dfu/package.py (hunk 2 above)
Comment (frkv) [reply nordicsemi/dfu/package.py]: Changing it

**Comment by frkv** [comment]
Closing as this is old

**Pull request closed without merging**